mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/dao/PostgresAttachmentDAO.java

```diff
@@ -80,9 +83,20 @@ public Mono<Pair<AttachmentMetadata, BlobId>> getAttachment(AttachmentId attachm
                 blobIdFactory.from(row.get(PostgresAttachmentTable.BLOB_ID))));
     }
 
+    public Flux<AttachmentMetadata> getAttachments(Collection<AttachmentId> attachmentIds) {
+        return postgresExecutor.executeRows(dslContext -> Flux.from(dslContext.selectFrom(PostgresAttachmentTable.TABLE_NAME)
```

Comment:
As my above commented
when using `IN` , we should partition it when `IN` size is large
I don't see any reply about it? wdyt

https://github.com/apache/james-project/pull/2048#discussion_r1502032559

Reply:
I think the number of attachments of a single message is not so big that we have to partition IN list

Reply:
fair enough